timedctl/timedctl.py

```diff
@@ -25,652 +24,527 @@
+            if task_data:
+                task = self.timed.tasks.get(id=task_data["id"], cached=True)
+            else:
+                task = {"attributes": {"name": "Unknown task"}, "id": None}
+            activity_view.append(
+                [
+                    task["attributes"]["name"],
+                    activity_obj["attributes"]["comment"],
+                    activity_obj["attributes"]["from-time"].strftime("%H:%M:%S")
+                    + " - "
+                    + activity_obj["attributes"]["to-time"].strftime("%H:%M:%S"),
+                    task["id"],
+                    activity_obj["id"],
+                ],
+            )
+        # get longest key per value
+        max_key_lengths = [
+            max(map(len, col)) for col in zip(*activity_view, strict=False)
+        ]
+        # pad all the values
+        activity_view = [
+            [
+                activity_view[i][j].ljust(max_key_lengths[j])
+                for j in range(len(activity_view[i]))
+            ]
+            for i in range(len(activity_view))
+        ]
+        # create a list for fzf
+        fzf_obj = []
+        for row in activity_view:
+            fzf_obj.append(
+                [" | ".join([row[0], row[1], row[2]]), row[1], row[2], row[3], row[4]],
+            )
+
+        activity_obj = fzf_wrapper(fzf_obj, [0], "Select an activity: ")
+        return activity_obj
+
+    def format_activity(self, activity_obj):
+        """Format an activity for display."""
+        task_obj = activity_obj["relationships"]["task"]
 
-    report = fzf_wrapper(fzf_obj, [0], "Select a report: ")
-    return report
+        task = task_obj["attributes"]["name"]
 
+        project_obj = self.timed.projects.get(
+            id=task_obj["relationships"]["project"]["id"],
+            cached=True,
+        )
+        project = project_obj["attributes"]["name"]
 
-def select_activity(date):
-    """FZF prompt to select an activity."""
-    activities = timed.activities.get(filters={"date": date})
-    activity_view = []
-    # loop through all activities
-    for activity_obj in activities:
-        # check if there is an actual task, else use an unknown task
-        task_data = activity_obj["relationships"]["task"]["data"]
-        if task_data:
-            task = timed.tasks.get(id=task_data["id"], cached=True)
+        customer_obj = self.timed.customers.get(
+            id=project_obj["relationships"]["customer"]["data"]["id"],
+            cached=True,
+        )
+        customer = customer_obj["attributes"]["name"]
+        return f"{customer} > {project} > {task}"
+
+    def get_customer_by_name(self, customers, name, archived):
+        """Get customer by name."""
+        customers = self.timed.customers.get(
+            cached=True,
+            filters={"archived": archived},
+        )
+        customer = [c for c in customers if c["attributes"]["name"] == name]
+        if len(customer) == 0:
```

Comment:
```suggestion
        if not customer:
```

Reply:
resolved as of https://github.com/adfinis/timedctl/pull/45/commits/13850fa95bec5246868b900b3db68602ca6edbf0